infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/InfinispanRemoteDialect.java

```diff
@@ -156,6 +166,26 @@ else if ( operation instanceof RemoveAssociationOperation ) {
 		owningEntity.flushOperations();
 	}
 
+	@Override
+	public ClosableIterator<Tuple> executeBackendQuery(BackendQuery<InfinispanRemoteQueryDescriptor> backendQuery, QueryParameters queryParameters, TupleContext tupleContext) {
+		return queryHandler.executeBackendQuery( backendQuery, queryParameters );
+	}
+
+	@Override
+	public int executeBackendUpdateQuery(BackendQuery<InfinispanRemoteQueryDescriptor> query, QueryParameters queryParameters, TupleContext tupleContext) {
+		throw new UnsupportedOperationException( "Update Query not supported by Infinispan Remote Dialect" );
+	}
+
+	@Override
+	public ParameterMetadataBuilder getParameterMetadataBuilder() {
+		return null;
+	}
+
+	@Override
+	public InfinispanRemoteQueryDescriptor parseNativeQuery(String nativeQuery) {
+		throw new UnsupportedOperationException( "Native Query not supported by Infinispan Remote Dialect" );
```

Comment:
In my first implementation I had used a String as query descriptor, like in Neo4j. But then I had to introduce the class InfinispanRemoteQueryDescriptor, because the dialect needs to know also the cache on which perform the query and the exact projection rows.

Reply:
What I mean is, why do you throw an exception. Can't we return the an InfinispanRemoteQueryDescriptor that represents the query instead of throwing an exception?  Does it take a lot of work? What do we need to check?

Reply:
Sorry I did not understand the question. Yes we can